dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -123,16 +124,51 @@ def write(self) -> None:
         from dlt.common.libs.deltalake import (
             write_delta_table,
             _deltalake_storage_options,
+            try_get_deltatable,
         )
 
         file_paths = [job.file_path for job in self.table_jobs]
 
-        write_delta_table(
-            path=self.client.make_remote_uri(self.make_remote_path()),
-            data=pa.dataset.dataset(file_paths),
-            write_disposition=self.table["write_disposition"],
-            storage_options=_deltalake_storage_options(self.client.config),
-        )
+        if (
+            self.table["write_disposition"] == "merge"
+            and (
+                dt := try_get_deltatable(
+                    self.client.make_remote_uri(self.make_remote_path()),
+                    storage_options=_deltalake_storage_options(self.client.config),
+                )
+            )
+            is not None
+        ):
+            assert self.table["x-merge-strategy"] in self.client.capabilities.supported_merge_strategies  # type: ignore[typeddict-item]
+
+            if self.table["x-merge-strategy"] == "upsert":  # type: ignore[typeddict-item]
+                if "parent" in self.table:
+                    unique_column = get_first_column_name_with_prop(self.table, "unique")
```

Comment:
Perhaps these column identifiers need to be escaped.

Reply:
just for my understanding: this first unique key will be the articifial row_hash created further above, right?

Reply:
The first `unique` key resolves to the `C_DLT_ID` column, which contains a key created by `_get_child_row_hash`.

https://github.com/dlt-hub/dlt/blob/e4d912eac07648cf713d6eabc5900a0a77ebb267/dlt/common/normalizers/json/relational.py#L506